### PR TITLE
(gh-519) Use MSI GUID to fix Remote Desktop pkg

### PR DESCRIPTION
**Pull request by ArcasCZ**
This fixes the broken Remote Desktop package install and uninstall process on non-English localised systems.

I've chosen to use `Get-WmiObject` to query for the information about app location for install purposes and localised name for uninstall purposes (as the returned object does not contain uninstall string).

### CODE REVIEW

File: automatic/remote-desktop-client/tools/chocolateyInstall.ps1 [hunk 1]
```diff
@@ -52,8 +52,12 @@ if (-Not (Test-Path -Path $registryPath)) {
...
 Set-ItemProperty -Path $registryPath -Name $name -Value $value -Type DWORD -Force | Out-Null
 
-$uninstallKey    = Get-UninstallRegistryKey -SoftwareName $packageArgs.SoftwareName
-$installLocation = $uninstallKey.InstallLocation
+
+# get install location based on the MSI GUID of the x64 OR x86 version of the app
+# x64 = 37482C7A-E958-455E-938E-0692B5C28708
+# x86 = 74E5B442-57F5-4097-95EA-38E8DBA1EBF1
+
+$installLocation = (Get-WmiObject -Class Win32_Product -Filter 'IdentifyingNumber like "{37482C7A-E958-455E-938E-0692B5C28708}" OR IdentifyingNumber like "{74E5B442-57F5-4097-95EA-38E8DBA1EBF1}"').InstallLocation
```
Comment (TheCakeIsNaOH) [inline automatic/remote-desktop-client/tools/chocolateyInstall.ps1:60]: FYI, `Get-WmiObject -Class Win32_Product` is not allowed on the community repository.

https://docs.chocolatey.org/en-us/community-repository/moderation/package-validator/rules/cpmr0059#mainContent

Comment (dgalbraith) [reply automatic/remote-desktop-client/tools/chocolateyInstall.ps1:60]: Given that `Get-WmiObject` is not permitted another approach is going to be needed here.

The installer is a single installer that contains all of the relevant localization files, and it looks like these are contained in specific country folders beneath the install location - eg. `zh-CN\msrdcw.resources.dll` which contain the localized resource files eg. `RdClient.WPF.Properties.Resources.zh-CN.resources`.

It is actually the `RemoteDesktopTitle` element that is being used to provide the localized element that is mapped to the `DisplayName` entry in the installation registry key.

```
  <data name="RemoteDesktopTitle" xml:space="preserve">
    <value>远程桌面</value>
  </data>
```

I would suggest extracting the `RemoteDesktopTitle` elements from the bundled resources and using these to build a lookup function to map the software name used in the `Get-UninstallRegistryKey` call appropriately.

Comment (ArcasCZ) [reply automatic/remote-desktop-client/tools/chocolateyInstall.ps1:60]: There's maybe another possible way to do this - registry query to HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\

**Comment by ArcasCZ** [comment]
@dgalbraith Any ideas / thoughts?

**Comment by ArcasCZ** [comment]
@dgalbraith Hi. Can we please move with this issue? It's a blocker for me.

**Comment by dgalbraith** [comment]
I don't think the GUIDs are install-neutral - they appear to be related to the localised install?  Looking at my install I have `{B39A7595-6140-4C59-A080-1344F25CF415}` for a 64-bit install on a `en-GB` host.

I am unable to locate 
```
# x64 = 37482C7A-E958-455E-938E-0692B5C28708
# x86 = 74E5B442-57F5-4097-95EA-38E8DBA1EBF1
```
associated with my installs.

**Comment by ArcasCZ** [comment]
There's simply no easy way to detect the product installed. I've also checked the MSI file with SuperOrca, but it does not contain any of the MSI GUID.

I've figured out that if I use `--install-args="'ProductLanguage=1033'"`, then I'm able to run the installer in EN_US language and also install the app that way without any trouble. (also, with EN_US lang it changes the GUID as well).

Closing the PR, but I'll keep the issue open. Maybe someone will figure some fix.

**Pull request closed without merging**